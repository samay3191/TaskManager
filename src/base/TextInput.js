import React from "react";
import PropTypes from "prop-types";
import DOMPurify from 'dompurify';

const TextInput = ({ value, onChange, placeHolder }) => {
  const handleChange = (e) => {
    const val = DOMPurify.sanitize(e.target.value);
    onChange(val);
  };

  return (
    <div className="m-1 leading-3">
      <input
        className="py-1 border border-gray-400 box-border outline-none rounded-lg px-2"
        type="text"
        value={value}
        onChange={handleChange}
        placeholder={placeHolder}
      />
    </div>
  );
};

TextInput.propTypes = {
  value: PropTypes.string.isRequired,
  placeHolder: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
};

export default TextInput;
