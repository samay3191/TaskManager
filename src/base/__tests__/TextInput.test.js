import React from "react";
import { shallow } from "enzyme";
import TextInput from "../TextInput";

const initialProps = {
  value: "",
  placeHolder: "Enter data",
  onChange: jest.fn(),
};

let component;

describe("<TextInput />", () => {
  beforeAll(() => {
    component = shallow(<TextInput {...initialProps} />);
  });

  it("should render TextInput component", () => {
    expect(component).toMatchSnapshot();
  });

  it("should render simulate TextInput change", () => {
    component.find("input").simulate("change", { target: { value: "a" }});
    expect(initialProps.onChange).toHaveBeenCalled();
  });
});
