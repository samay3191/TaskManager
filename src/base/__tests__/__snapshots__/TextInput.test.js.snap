// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextInput /> should render TextInput component 1`] = `
<div
  className="m-1 leading-3"
>
  <input
    className="py-1 border border-gray-400 box-border outline-none rounded-lg px-2"
    onChange={[Function]}
    placeholder="Enter data"
    type="text"
    value=""
  />
</div>
`;
